### PR TITLE
Rebenchmark pallet-mantapay

### LINKED ISSUE #1146: Rebenchmark pallet manta pay

### PR DESCRIPTION
**Pull request by Dengjianping**
## Description

Closes #1146 

---

Before we can approve this PR for merge, please make sure that **all** the following items have been checked off:
- [x] Connected to an issue with discussion and accepted design using zenhub "Connect issue" button below
- [x] Added **one** label out of the `L-` group to this PR
- [x] Added **one or more** labels from the `A-` and `C-` groups to this PR
- [x] Explicitly labelled `A-calamari` and/or `A-manta` if your changes are meant for/impact either of these (CI depends on it)
- [ ] Re-reviewed `Files changed` in the Github PR explorer.

Situational Notes:
- If adding functionality, write unit tests!
- If importing a new pallet, choose a proper module index for it, and allow it in `BaseFilter`. Ensure **every** extrinsic works from front-end. If there's corresponding tool, ensure both work for each other.
- If needed, update our Javascript/Typescript APIs. These APIs are officially used by exchanges or community developers.
- If modifying existing runtime storage items, make sure to implement storage migrations for the runtime and test them with `try-runtime`. This includes migrations inherited from upstream changes, and you can search the diffs for modifications of `#[pallet::storage]` items to check for any.

### CODE REVIEW

File: pallets/manta-pay/src/benchmark/mod.rs [hunk 1]
```diff
@@ -28,16 +23,29 @@ use manta_support::manta_pay::{
...
+const MINTS_OFFSET: usize = 2;
+const MINT_SIZE: usize = 553;
+
+const TRANSFERS_OFFSET: usize = 4;
+const TRANSFER_SIZE: usize = 1291;
+
+const RECLAIMS_OFFSET: usize = 4;
+const RECLAIM_SIZE: usize = 1001;
+
+// 14000 coins have been inserted into the chain spec,
```
Comment (ghzlatarev) [inline pallets/manta-pay/src/benchmark/mod.rs]: 14000 iterations of mint, mint, mint, transfer, mint, mint, reclaim

File: pallets/manta-pay/src/benchmark/mod.rs [hunk 2]
```diff
@@ -28,16 +23,29 @@ use manta_support::manta_pay::{
...
 use manta_primitives::{
-    assets::{AssetConfig, AssetRegistry, FungibleLedger, TestingDefault},
+    assets::{AssetConfig, FungibleLedger},
     constants::TEST_DEFAULT_ASSET_ED,
     types::Balance,
 };
 use scale_codec::Decode;
 
-mod precomputed_coins;
+// mod precomputed_coins;
```
Comment (ghzlatarev) [inline pallets/manta-pay/src/benchmark/mod.rs]: commented code

**Review by ghzlatarev — COMMENTED** [review]
looking good, let's run it and see if we get different results. i have a feeling we'll need more coins.

File: pallets/manta-pay/src/benchmark/mod.rs (hunk 2 above)
Comment (Dengjianping) [reply pallets/manta-pay/src/benchmark/mod.rs]: Should i delete this `precomputed_coins.rs` file? If this pr gets merged, seems this file is unnecessary.

Comment (ghzlatarev) [reply pallets/manta-pay/src/benchmark/mod.rs]: yeah seems like it

**Review by ferrell-code — COMMENTED** [review]
CI error in benchmarks, Looks reasonable  to me curious what new benchmark result will be. 

We also need to do the same for MantaSbt, but that can be different PR

**Comment by Dengjianping** [comment]
> CI error in benchmarks, Looks reasonable to me curious what new benchmark result will be.
> 
> We also need to do the same for MantaSbt, but that can be different PR

#1151

**Comment by Dengjianping** [comment]
calamari benchmark result: https://github.com/Manta-Network/Manta/actions/runs/5264802285/jobs/9516655255
manta benchmark result: https://github.com/Manta-Network/Manta/actions/runs/5266279560/jobs/9523638319

File: .github/workflows/run_all_benchmarks.yml [hunk 3]
```diff
@@ -77,19 +84,31 @@ jobs:
...
+      - name: create-chainspec
+        run: |
+          ${{ github.workspace }}/target/production/manta build-spec --chain $CHAIN_SPEC --disable-default-bootnode --raw > ${{ github.workspace }}/tests/data/fork.json
+      - name: append manta-pay storage
+        run: |
+          wget -P ${{ github.workspace }}/tests/data https://manta-ops.s3.amazonaws.com/integration-tests-data/storage.json
+          cd ${{ github.workspace }}/tests
+          yarn install
+          yarn
+          node append_storage.js
```
Comment (ferrell-code) [inline .github/workflows/run_all_benchmarks.yml:96]: are we sure this appends to `Manta/tests/data/fork.json` and not `Manta/data/fork.json`

Comment (ghzlatarev) [reply .github/workflows/run_all_benchmarks.yml:96]: there's a `cd tests/` command

**Review by ghzlatarev — COMMENTED** [review]
This actually breaks the mantapay benchmarks units tests

Not sure if there's a way to just disable the unit tests for MantaPay benchmark ?

**Comment by Dengjianping** [comment]
> 

Because rebenchmarking mantapay uses a concrete chain spec, which contains some storage that will affect benchmark code, like there're some asset ids registered already, so we cannot register them in benchmark code.

**Comment by ghzlatarev** [comment]
> > 
> 
> Because rebenchmarking mantapay uses a concrete chain spec, which contains some storage that will affect benchmark code, like there're some asset id registered already, so we cannot register in benchmark code.

right can we just exclude MantaPay from our benchmark test in check-tests workflow

**Review by ghzlatarev — DISMISSED** [review]

**Review by ghzlatarev — DISMISSED** [review]

**Review by ghzlatarev — APPROVED** [review]

**Review by Apokalip — APPROVED** [review]

**Pull request merged**